lib/msf/core/post/linux/system.rb

```diff
@@ -96,6 +96,33 @@ def get_sysinfo
     return system_data
   end
 
+  #
+  # return true if possible chroot environment detect, if not true then false
+  #
+  def is_chroot?
+    chroot_env = false
+
+    if not directory?("/proc/")
+      chroot_env = true
+    else
+      dir("/proc/").each do |pid|
+        pid = Integer(pid) rescue nil
+        if pid
+          filename = ::File.readlink("/proc/#{pid}/exe") rescue nil
```

Comment:
@void-in I need check with the file represented by the symlink exist, if I use File.exists all the file will be return success, even if no exists, and I think this also operates locally, like File.readlink